src/main/java/net/neoforged/neoforge/event/entity/player/PlayerEvent.java

```diff
@@ -406,6 +406,37 @@ public Container getInventory() {
         }
     }
 
+    public static class ItemSmithingEvent extends PlayerEvent {
```

Comment:
I would like to see some documentation on this class. Most of our events should have a javadoc explaining:
- What the event's purpose is (e.g. "An event for when a player smiths an item")
- When the event is fired (e.g. "After awarding used recipes, but before resizing the stacks")
- *Where* the event is fired (e.g. on the logical server, client, both)
- Which bus it's fired on (e.g. the game bus)

Reply:
Understood, i will see to this asap

Reply:
This should be added!